### PR TITLE
Mock auth screen

### PR DESCRIPTION
**Pull request by Ionnier**

### CODE REVIEW

File: app/src/main/java/com/raduitache/myanimelist/anime_details/AnimeDetailsScreen.kt [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            Box(modifier = Modifier.fillMaxSize()
+                .clickable(interactionSource = remember {
+                    MutableInteractionSource()
+                }, indication = null, onClick = {
+                    selectedImage = null
+                }).background(
+                Color.Black.copy(alpha = 0.8f))) {
+                AsyncImage(model = it, contentDescription = null, modifier = Modifier.fillMaxSize())
+            }
+
```
Comment (raduitache) [inline app/src/main/java/com/raduitache/myanimelist/anime_details/AnimeDetailsScreen.kt:136]: Remove extra lines

File: app/src/main/java/com/raduitache/myanimelist/anime_details/AnimeDetailsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+@HiltViewModel
+class AnimeDetailsViewModel @Inject constructor(
+    private val animeServiceFlow: Flow<@JvmSuppressWildcards AnimeService>
+) : ViewModel() {
+    private val _animeDetailsScreenState = MutableStateFlow(AnimeDetailsScreenState())
+    val animeDetailsScreenState = _animeDetailsScreenState.asStateFlow()
+
+    fun selectAnime(id: String) {
+        viewModelScope.launch {
+            val service = animeServiceFlow.firstOrNull() ?: return@launch
```
Comment (raduitache) [inline app/src/main/java/com/raduitache/myanimelist/anime_details/AnimeDetailsViewModel.kt]: set _animeDetalisScreenState isLoading to true before these blocking requests.

File: app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+                .padding(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            item {
+                Box(
+                    modifier = Modifier
+                        .size(width = 128.dp, height = 64.dp)
+                        .background(Color.Red)
+                )
+                Box(modifier = Modifier.heightIn(min = 120.dp))
```
Comment (raduitache) [inline app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt:89]: What's the point of these?

File: app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+                        val transformedText = text.toString()
+                            .mapIndexed { index, c ->
+                                "*"
+                            }
+                            .joinToString("")
+                        TransformedText(
+                            text = AnnotatedString(text = transformedText),
+                            offsetMapping = OffsetMapping.Identity
+                        )
+                    },
```
Comment (raduitache) [inline app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt]: No need to do this by hand, use [this](https://developer.android.com/reference/kotlin/androidx/compose/ui/text/input/PasswordVisualTransformation).

File: app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+                if (state.shouldNavigateNext) {
+                    Spacer(Modifier.size(4.dp))
+                    Text(
+                        text = "Operation successful, you will be redirected shortly",
+                        style = TextStyle(color = MaterialTheme.colorScheme.secondary)
+                    )
+                    LaunchedEffect(key1 = Unit) {
+                        delay(1500)
+
```
Comment (raduitache) [inline app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt]: nit: delete empty line

File: app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+                    ) {
+                        Text(user.username, modifier = Modifier.weight(1f))
+                        Text(user.email, modifier = Modifier.weight(1f))
+                        Text(
+                            user.password,
+                            overflow = TextOverflow.Ellipsis,
+                            maxLines = 1, modifier = Modifier.weight(1f)
+                        )
+                    }
+
```
Comment (raduitache) [inline app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt]: nit: delete empty line

File: app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+                }
+            }
+
```
Comment (raduitache) [inline app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt]: nit: delete empty line

File: app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+        }
+
+
```
Comment (raduitache) [inline app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt]: nit: delete empty lines.

File: app/src/main/java/com/raduitache/myanimelist/seasonal/impl/SeasonalNavGraphImpl.kt [hunk 9]
```diff
@@ -37,7 +43,20 @@ class SeasonalNavGraphImpl @Inject constructor(
...
+        composable(
+            "details/{details}",
+            arguments = listOf(navArgument("details") {type = NavType.StringType})
+        ) { backStackEntry ->
+            val animeDetailsViewModel = hiltViewModel<AnimeDetailsViewModel>()
+            backStackEntry.arguments?.getString("details")?.let { animeDetailsViewModel.selectAnime(it) }
+            AnimeDetailsScreen(animeDetailsViewModel) {
+                navController.popBackStack()
+            }
+        }
```
Comment (raduitache) [inline app/src/main/java/com/raduitache/myanimelist/seasonal/impl/SeasonalNavGraphImpl.kt:60]: I'd have this in a different navGraph, which is not a main nav graph. The bottom bar navigation shouldn't be visible when in an anime details view. Feel free to keep it as is for now, though.

**Review by raduitache — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt (hunk 3 above)
Comment (Ionnier) [reply app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt:89]: LASA-MA VERE IN PACE CAT TIMP MERGE NU DAM IN PRODUCTIE

File: app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt (hunk 8 above)
Comment (Ionnier) [reply app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt]: fgerjkgjdreklfghjerdflkgdrefj

File: app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt (hunk 3 above)
Comment (Ionnier) [reply app/src/main/java/com/raduitache/myanimelist/auth/impl/AuthNavRouteImpl.kt:89]: era un placeholder ca asa era un wireframe, in loc de poza

**Review by raduitache — APPROVED** [review]

**Pull request merged**